### PR TITLE
defaultstorageclass: Enable metrics collection

### PR DESCRIPTION
**Pull request by gracedo**
**What type of PR is this?**

Bug

**What this PR does/ why we need it**:

Annotation-based service discovery has been deprecated in prometheus-operator so these metrics were no longer getting collected. Switch to use the service monitor model by adding this label to allow it to get scraped by Prometheus. This requires a change to the Prometheus addon config to add this metrics endpoint (will put up kba pr for this later after including all other necessary prom config changes to get metrics collected from other components):

```diff
--- a/addons/prometheus/prometheus.yaml
+++ b/addons/prometheus/prometheus.yaml
@@ -100,6 +100,24 @@ spec:
               # configuration.
               - targetPort: 10902
                 interval: 30s
+          - name: kubeaddons-service-monitor-metrics-defaultstorageclass
+            selector:
+              matchLabels:
+                servicemonitor.kubeaddons.mesosphere.io/path: "metrics"
+                kubeaddons.mesosphere.io/name: "defaultstorageclass"
+            namespaceSelector:
+              matchNames:
+                - kubeaddons
+            endpoints:
+              - port: https
+                interval: 30s
+                scheme: https
+                bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+                tlsConfig:
+                  caFile: "/etc/prometheus/secrets/dstorageclass-webhook-server-cert/ca.crt"
+                  certFile: "/etc/prometheus/secrets/dstorageclass-webhook-server-cert/tls.crt"
+                  keyFile: "/etc/prometheus/secrets/dstorageclass-webhook-server-cert/tls.key"
+                  insecureSkipVerify: true
           - name: kubeaddons-service-monitor-api-v1-metrics-prometheus
             selector:
               matchLabels:
@@ -243,6 +261,7 @@ spec:
           walCompression: true
           secrets:
             - etcd-certs
+            - dstorageclass-webhook-server-cert
           externalUrl: "/ops/portal/prometheus"
           storageSpec:
             volumeClaimTemplate:
```

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-72691

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

* [x] *If a chart is changed, the chart version is correctly incremented.*
* [x] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Comment by gracedo** [comment]
Tested this manually, saw the target come up healthy
image

File: stable/defaultstorageclass/templates/service.yaml [hunk 1]
```diff
@@ -2,13 +2,11 @@
...
 metadata:
-  annotations:
-    prometheus.io/port: "8443"
-    prometheus.io/scheme: https
-    prometheus.io/scrape: "true"
   creationTimestamp: null
   labels:
     control-plane: controller-manager
+    servicemonitor.kubeaddons.mesosphere.io/path: metrics
+    kubeaddons.mesosphere.io/name: defaultstorageclass
```
Comment (gracedo) [inline stable/defaultstorageclass/templates/service.yaml:9]: I'm not in love with this label. Open to using other unique identifying labels for this chart/addon

Comment (hectorj2f) [reply stable/defaultstorageclass/templates/service.yaml:9]: Yes, I am not either.

**Review by hectorj2f — CHANGES_REQUESTED** [review]
@gracedo there is another way such as reusing one of the existing scraping configurations for addons. The solution would be to adapt the port-name to be named metrics. Let's sync later.

**Comment by gracedo** [comment]
> @gracedo there is another way such as reusing one of the existing scraping configurations for addons. The solution would be to adapt the port-name to be named metrics. Let's sync later.

The issue I ran into here (and with dex-controller) was that both these endpoints are https and needed tlsConfigs. I could reuse the metrics servicemonitor definition as is by moving this entire endpoint into that def, but dex-controller uses the same port/name so both endpoints would be targeted, however they have different certs. We could rename the port in both so they are unique, but then I’m not sure if that is any better than defining two servicemonitors. I can be persuaded tho :) and maybe there is another workaround I haven’t thought of which is likely!

**Comment by hectorj2f** [comment]
For systems such as dex-controller, I won't touch it and add this mechanism you used here (it is better to keep the service and do not restart it). For this chart, we can simply change the metrics port, it has no major impact to expose these metrics of this chart.

**Comment by hectorj2f** [comment]
@gracedo For the sake of fixing the issue now, let's merge this approach.

**Review by hectorj2f — APPROVED** [review]

**Review by branden — APPROVED** [review]

**Pull request merged**